### PR TITLE
Merge .config & nni.sqlite

### PR DESCRIPTION
**Pull request by J-shang**

### CODE REVIEW

File: nni/tools/nnictl/nnictl_utils.py [hunk 1]
```diff
@@ -986,43 +991,42 @@ def load_experiment(args):
...
-            exit(1)
-        else:
-            for file in os.listdir(archive_search_space_dir):
-                source_path = os.path.join(archive_search_space_dir, file)
-                os.makedirs(os.path.dirname(target_path), exist_ok=True)
-                shutil.copyfile(source_path, target_path)
-                break
-    elif not args.searchSpacePath:
-        print_warning('%s exist, will not load search_space file' % target_path)
+    # archive_search_space_dir = os.path.join(temp_root_dir, 'searchSpace')
```
Comment (SparkSnail) [inline nni/tools/nnictl/nnictl_utils.py:994]: Why annotate them?

Comment (J-shang) [reply nni/tools/nnictl/nnictl_utils.py]: yes, search space file does not need a copy anymore, have deleted them.

File: nni/tools/nnictl/config_utils.py [hunk 2]
```diff
@@ -4,64 +4,106 @@
...
 import json
 import shutil
+import sqlite3
 import time
-from .constants import NNICTL_HOME_DIR
+from .constants import NNI_HOME_DIR
 from .command_utils import print_error
 from .common_utils import get_file_lock
 
+def config_v0_to_v1(config: dict) -> dict:
```
Comment (SparkSnail) [inline nni/tools/nnictl/config_utils.py:13]: What does `v0` mean?

Comment (J-shang) [reply nni/tools/nnictl/config_utils.py:13]: v0 means the config format in nni.sqlite (clusterMetaData), will be deprecated soon.

File: nni/tools/nnictl/config_utils.py [hunk 3]
```diff
@@ -4,64 +4,106 @@
...
+        self.refresh_config()
 
-    def get_all_config(self):
-        '''get all of config values'''
-        return json.dumps(self.config, indent=4, sort_keys=True, separators=(',', ':'))
+    def refresh_config(self):
+        '''refresh to get latest config'''
+        sql = 'select * from ExperimentProfile where id=? order by revision DESC'
+        args = (self.experiment_id,)
+        self.config = config_v0_to_v1(json.loads(self.cursor.execute(sql, args).fetchall()[0][0]))
```
Comment (liuzhe-lz) [inline nni/tools/nnictl/config_utils.py:90]: Don't use `self.cursor`. One cursor per query.

Comment (liuzhe-lz) [reply nni/tools/nnictl/config_utils.py:90]: And `.fetchall()[0]` should be `.fetchone()`.

File: nni/tools/nnictl/launcher.py [hunk 4]
```diff
@@ -624,26 +618,25 @@ def create_experiment(args):
 def manage_stopped_experiment(args, mode):
     '''view a stopped experiment'''
     update_experiment()
-    experiment_config = Experiments()
-    experiment_dict = experiment_config.get_all_experiments()
+    experiments_config = Experiments()
```
Comment (liuzhe-lz) [inline nni/tools/nnictl/launcher.py:621]: I personally suggest rename "config" to "metadata" or "metainfo".
Not mandatory if you think "config" is better.

File: nni/tools/nnictl/config_utils.py (hunk 3 above)
Comment (J-shang) [reply nni/tools/nnictl/config_utils.py:90]: fix

**Review by liuzhe-lz — APPROVED** [review]

**Pull request merged**